python/paddle/tensor/math.py

```diff
@@ -217,11 +233,13 @@ def _elementwise_op(helper):
     axis = helper.kwargs.get('axis', -1)
     use_mkldnn = helper.kwargs.get('use_mkldnn', False)
     name = helper.kwargs.get('name', None)
-    if name is None:
-        out = helper.create_variable_for_type_inference(dtype=x.dtype)
+    if out_tmp is not None:
```

Comment:
You can write code short:

change line 222:
```
out = helper.kwargs.get('out', None)
```
then line 236 ~ line 242 will be:

```
if out is None:
     if name is None:
          XXX
     else:
          XXX
```

In this case 1. you remove a useless python variable out_tmp, 2. you remove an else case.

Code is simpler.